### PR TITLE
Open different socket to enable interface filtering for dual tor scenario

### PR DESCRIPTION
**Pull request by kellyyeh**
Why I did it
Raw sockets are used for receiving packets and acl rules are not able to block dhcpv6 packets from being received by dhcp6relay raw sockets.

How I did it
To accommodate dual tor scenario, recvfrom is used to filter out interfaces. If the interface mux status is standby, dhcp6relay does not relay the packet.

### CODE REVIEW

File: src/relay.cpp [hunk 1]
```diff
@@ -785,63 +794,71 @@ void dhcp6relay_stop()
...
         event_add(listen_event, NULL);
         event_add(server_listen_event, NULL);
-    }
-    
-    if((signal_init() == 0) && signal_start() == 0) {
-        shutdown();
-        for(std::size_t i = 0; i<sockets.size(); i++) {
-            close(sockets.at(i));
+        
+        if((signal_init() == 0) && signal_start() == 0) {
```
Comment (saiarcot895) [inline src/relay.cpp:857]: Should this be inside a for-loop? I think this should only be run once.

Comment (kellyyeh) [reply src/relay.cpp]: Moved it outside of for loop

File: src/main.cpp [hunk 2]
```diff
@@ -2,12 +2,30 @@
 #include <syslog.h>
 #include "configInterface.h"
 
+bool dual_tor_sock = false;
+
+static void usage()
+{
+    printf("Usage: ./dhcp6relay {-d}\n");
+    printf("\t-u: enable dual tor option\n");
```
Comment (saiarcot895) [inline src/main.cpp:10]: The previous line has `-d`, but this line has `-u`.

**Comment by saiarcot895** [comment]
With this change, `src/relay.cpp` will be almost 1000 lines. I think it might be worth it for maintainability purposes to reorganize this file in the future. Maybe split it into separate files, or use C++ classes to group functions and structures used?

File: src/main.cpp (hunk 2 above)
Comment (kellyyeh) [reply src/main.cpp]: Thanks, updated

File: src/relay.cpp [hunk 3]
```diff
@@ -18,29 +18,34 @@ struct event *server_listen_event;
...
+    { 0x20, 0, 0, 0x0000002e },
+    { 0x15, 0, 7, 0x00000000 },
+    { 0x20, 0, 0, 0x00000032 },
+    { 0x15, 0, 5, 0x00010002 },
+    { 0x30, 0, 0, 0x00000014 },
+    { 0x15, 0, 3, 0x00000011 },
+    { 0x28, 0, 0, 0x00000038 },
+    { 0x15, 0, 1, 0x00000223 },
+    { 0x6, 0, 0, 0x00040000 },
+    { 0x6, 0, 0, 0x00000000 },
```
[the remark this answers is not in the document]
Comment (kellyyeh) [reply src/relay.cpp:47]: Thanks, updated

**Comment by kellyyeh** [comment]
> With this change, `src/relay.cpp` will be almost 1000 lines. I think it might be worth it for maintainability purposes to reorganize this file in the future. Maybe split it into separate files, or use C++ classes to group functions and structures used?

Agreed, I will create another PR for this.

**Review by yxieca — APPROVED** [review]

**Review by saiarcot895 — APPROVED** [review]

**Pull request merged**